### PR TITLE
Private source file object used, instead of com.sun.tools.javac types

### PR DESCRIPTION
**Pull request by sormuras**
Proposal for #391 - all tests remain green.

Is the internal "Source" class too simplistic? Am I missing something?

### CODE REVIEW

File: src/test/java/com/squareup/javapoet/TestFiler.java [hunk 1]
```diff
@@ -30,20 +31,28 @@
...
+import javax.tools.SimpleJavaFileObject;
 
 final class TestFiler implements Filer {
+  static class Source extends SimpleJavaFileObject {
+    private final OutputStream stream;
+    protected Source(URI uri, OutputStream stream) {
+      super(uri, Kind.SOURCE);
+      this.stream = stream;
+    }
+    @Override public OutputStream openOutputStream() {
```
Comment (swankjesse) [inline src/test/java/com/squareup/javapoet/TestFiler.java]: I think this method probably wants to create a new stream on every invocation?

Comment (swankjesse) [inline src/test/java/com/squareup/javapoet/TestFiler.java]: it's weird to have a live object like a stream as a field of a value object

**Comment by swankjesse** [comment]
Nice!

File: src/test/java/com/squareup/javapoet/TestFiler.java (hunk 1 above)
Comment (sormuras) [inline src/test/java/com/squareup/javapoet/TestFiler.java]: As an instance of `Source` is only created by `TestFiler#createSourceFile`, I think it's okay to remember the already opened stream here. That instance will be (and according to all the green lights *is*) only used to collect the Java source lines once. I'm 99% sure, that `openOutputStream` is only called a single time per `TestFiler#createSourceFile`.

Comment (sormuras) [inline src/test/java/com/squareup/javapoet/TestFiler.java]: I move the directory, file and stream creation into `Source`. Feels the 1% better.

**Pull request merged**